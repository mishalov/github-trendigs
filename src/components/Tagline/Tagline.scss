@import "../../styles/variables.scss";

.tagline {
  margin-top: 16px;
  font-size: 12px;
  display: flex;

  &__item {
    img {
      margin-right: 6px;
    }
  }
  &__item:not(:last-child) {
    padding-right: 8px;
    border-right: 1px solid $dark-border-color;
  }

  &__item:not(:first-child) {
    padding-left: 8px;
  }

  &__stargazers {
    display: flex;
    align-items: center;
  }
}
