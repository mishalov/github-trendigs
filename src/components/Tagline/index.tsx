import React from "react";
import starBlack from "../../assets/svg/star_black.svg";
import fork from "../../assets/svg/fork.svg";
import "./Tagline.scss";
import { divideNumbers } from "../../utils/divideNumbers";

interface ITaglineProps {
  language: string | null;
  stargazersCount: number;
  forks: number;
}

/**
 * Component to show meta info of repository in Repo list
 */
const Tagline: React.FC<ITaglineProps> = props => {
  const { language, stargazersCount, forks } = props;
  return (
    <div className="tagline">
      {language && (
        <div
          className="tagline__item"
          title="Programming language / technologies"
        >
          {language}
        </div>
      )}
      <div
        className="tagline__item tagline__stargazers"
        title="Total stargazers"
      >
        <img src={starBlack} width="12" height="12" alt="star" />
        {divideNumbers(stargazersCount)}
      </div>
      <div className="tagline__item" title="Forks">
        <img src={fork} width="12" height="12" alt="forks" />
        {divideNumbers(forks)}
      </div>
    </div>
  );
};

export default Tagline;
