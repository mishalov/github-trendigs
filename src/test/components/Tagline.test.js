import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import Tagline from "../../components/Tagline";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

const language = "python";
const languageNulled = null;
const stargazersCount = 100;
const forks = 150;

beforeEach(() => {
  wrappers.push(
    shallow(
      <Tagline
        language={language}
        stargazersCount={stargazersCount}
        forks={forks}
      />
    )
  );
  wrappers.push(
    shallow(
      <Tagline
        language={languageNulled}
        stargazersCount={stargazersCount}
        forks={forks}
      />
    )
  );
});

describe("<Loader /> rendering", () => {
  it("renders correctly with language", () => {
    expect(wrappers[0]).toMatchSnapshot();
  });
  it("renders correctly without language", () => {
    expect(wrappers[1]).toMatchSnapshot();
  });
});
