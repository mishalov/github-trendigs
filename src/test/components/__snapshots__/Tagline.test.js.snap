// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> rendering renders correctly with language 1`] = `
<div
  className="tagline"
>
  <div
    className="tagline__item"
    title="Programming language / technologies"
  >
    python
  </div>
  <div
    className="tagline__item tagline__stargazers"
    title="Total stargazers"
  >
    <img
      alt="star"
      height="12"
      src="star_black.svg"
      width="12"
    />
    100
  </div>
  <div
    className="tagline__item"
    title="Forks"
  >
    <img
      alt="forks"
      height="12"
      src="fork.svg"
      width="12"
    />
    150
  </div>
</div>
`;

exports[`<Loader /> rendering renders correctly without language 1`] = `
<div
  className="tagline"
>
  <div
    className="tagline__item tagline__stargazers"
    title="Total stargazers"
  >
    <img
      alt="star"
      height="12"
      src="star_black.svg"
      width="12"
    />
    100
  </div>
  <div
    className="tagline__item"
    title="Forks"
  >
    <img
      alt="forks"
      height="12"
      src="fork.svg"
      width="12"
    />
    150
  </div>
</div>
`;
